pkg/cephmgr/osd/partition/scheme.go

```diff
@@ -32,78 +33,96 @@ const (
+// and entries for all OSDs that define where their partitions live
+type PerfScheme struct {
+	Metadata *MetadataDeviceInfo `json:"metadata"`
+	Entries  []*PerfSchemeEntry  `json:"entries"`
+}
+
+// represents an OSD and details about all of its partitions
+type PerfSchemeEntry struct {
+	ID         int                                    `json:"id"`
+	OsdUUID    uuid.UUID                              `json:"osdUuid"`
+	Partitions map[string]*PerfSchemePartitionDetails `json:"partitions"` // mapping of partition name to its details
+}
+
+// details for 1 OSD partition
+type PerfSchemePartitionDetails struct {
+	Device        string `json:"device"`
+	DiskUUID      string `json:"diskUuid"`
+	PartitionUUID string `json:"partitionUuid"`
+	SizeMB        int64  `json:"sizeMb"`
+	Offset        int64  `json:"offset"`
+}
+
+// represents a dedicated metadata device and all of the partitions stored on it
+type MetadataDeviceInfo struct {
+	Device     string                     `json:"device"`
+	DiskUUID   string                     `json:"diskUuid"`
+	Partitions []*MetadataDevicePartition `json:"partitions"`
+}
+
+// representsa specific partition on a metadata device, including details about which OSD it belongs to
+type MetadataDevicePartition struct {
+	ID            int       `json:"id"`
+	OsdUUID       uuid.UUID `json:"osdUuid"`
+	Name          string    `json:"name"`
+	PartitionUUID string    `json:"partitionUuid"`
+	SizeMB        int64     `json:"sizeMb"`
+	Offset        int64     `json:"offset"`
+}
+
+func NewPerfScheme() *PerfScheme {
+	return &PerfScheme{
+		Entries: []*PerfSchemeEntry{},
 	}
+}
 
-	uuids := map[string]string{
-		WalPartitionName:      walUUID.String(),
-		DatabasePartitionName: dbUUID.String(),
-		BlockPartitionName:    blockUUID.String(),
+func NewPerfSchemeEntry() *PerfSchemeEntry {
+	return &PerfSchemeEntry{
+		Partitions: map[string]*PerfSchemePartitionDetails{}, // mapping of partition name (e.g. WAL) to it's details
 	}
+}
 
-	return &Scheme{
-		Version:        SimpleVersion,
-		SizeMB:         sizeMB,
-		ID:             id,
-		DiskUUID:       diskUUID.String(),
-		PartitionUUIDs: uuids,
-	}, nil
+func NewMetadataDeviceInfo(device string) *MetadataDeviceInfo {
+	return &MetadataDeviceInfo{Device: device, Partitions: []*MetadataDevicePartition{}}
 }
 
-// Turn the scheme into arguments for sgdisk. The result will be three partitions:
-// 1. WAL: Write ahead log (10%)
-// 2. DB: The bluestore database (10%)
-// 3. Block: The raw storage for the data being written to the OSD. (80%)
-//     Uses the remainder of the storage space after small partitions for the wal and db.
-func (s *Scheme) GetArgs(name string) []string {
-	offset := 1
-	tenPercent := s.SizeMB / 10
+// Load the persistent partition info from the config directory.
+func LoadScheme(configDir string) (*PerfScheme, error) {
+	filePath := path.Join(configDir, schemeFilename)
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
```

Comment:
good question, yes we'll fall though and attempt the read, which would likely error also and then that error would be returned.